src/clib/pio_file.c

```diff
@@ -113,35 +147,45 @@ int PIOc_openfile(const int iosysid, int *ncidp, int *iotype,
 
     // If we failed to open a file due to an incompatible type of NetCDF, try it 
     // once with just plain old basic NetCDF
-#ifdef _NETCDF
-    if((ierr == NC_ENOTNC || ierr == NC_EINVAL) && (file->iotype != PIO_IOTYPE_NETCDF)) {
-        if(ios->iomaster) printf("PIO2 pio_file.c retry NETCDF\n"); 
-	// reset ierr on all tasks
-	ierr = PIO_NOERR;
-	// reset file markers for NETCDF on all tasks
-	file->iotype = PIO_IOTYPE_NETCDF;
-
-	// open netcdf file serially on main task
-        if(ios->io_rank==0){
-	  ierr = nc_open(filename, file->mode, &(file->fh)); }
-
-    }
-#endif
+/* #ifdef _NETCDF */
```

Comment:
I have uncommented these lines of code again, as you request.

My intention is NOT to do this with the new PIOc_open call. If this kind of format change is going to take place, it should be in the user's code, where they take full responsibility for the format they are using.

If we bury netcdf or data bugs under code like this, then they will never get fixed.